spec/truffle/java_exception_spec.rb

```diff
@@ -22,11 +22,14 @@
     lambda { Truffle::Debug.throw_java_exception 'message' }.should raise_error { |e|
       message = e.message.gsub(/:\d+/, ':LINE')
       
-      # This line can vary depending on specialization order
-      message.lines[2].start_with?("\torg.truffleruby.debug.TruffleDebugNodesFactory$ThrowJavaExceptionNodeFactory$ThrowJavaExceptionNodeGen.execute").should be_true
-      message.lines[2].end_with?("(TruffleDebugNodesFactory.java:LINE)\n").should be_true
-      
-      message.lines[3].should == "\torg.truffleruby.language.control.SequenceNode.execute(SequenceNode.java:LINE)\n"
+      if message.lines[2].include?("executeAndSpecialize")
+        message.lines[2].should == "\torg.truffleruby.debug.TruffleDebugNodesFactory$ThrowJavaExceptionNodeFactory$ThrowJavaExceptionNodeGen.executeAndSpecialize(TruffleDebugNodesFactory.java:LINE)\n"
+        message.lines[3].should == "\torg.truffleruby.debug.TruffleDebugNodesFactory$ThrowJavaExceptionNodeFactory$ThrowJavaExceptionNodeGen.execute(TruffleDebugNodesFactory.java:LINE)\n"
+        message.lines[4].should == "\torg.truffleruby.language.control.SequenceNode.execute(SequenceNode.java:LINE)\n"
+      else
+        message.lines[2].should == "\torg.truffleruby.debug.TruffleDebugNodesFactory$ThrowJavaExceptionNodeFactory$ThrowJavaExceptionNodeGen.execute(TruffleDebugNodesFactory.java:LINE)\n"
+        message.lines[3].should == "\torg.truffleruby.language.control.SequenceNode.execute(SequenceNode.java:LINE)\n"
```

Comment:
Maybe we should just have the exception thrown in its own java file through a couple methods and we assert those? It would be more robust.